sydent/sydent.py

```diff
@@ -307,30 +307,45 @@ def parse_config_dict(config_dict):
     Args:
         config_dict (dict): the configuration dictionary to be parsed
     """
-
+    # Build a config dictionary from the defaults merged with the given dictionary
+    config = CONFIG_DEFAULTS
+    for section, section_dict in config_dict.items():
+        if section not in config:
+            config[section] = {}
+        for option in section_dict.keys():
+            config[section][option] = config_dict[section][option]
```

Comment:
this stuff is in a commit called "Merge branch 'master' into babolivier/py3" but afaict it's not on master yet. I'm therefore confused about where it comes from and what it is doing?

Reply:
That was merging [these changes](https://github.com/matrix-org/sydent/pull/141/files#diff-39f6d4aebe64fc43892acd9551a558d3R303-R348) into my branch and fixing the type for the ConfigParser.